docarray/doc_index/backends/weaviate_doc_index.py

```diff
@@ -0,0 +1,594 @@
+
+    def _create_schema(self) -> None:
+        schema = {}
+
+        properties = []
+        column_infos = self._column_infos
+
+        for column_name, column_info in column_infos.items():
+            # in weaviate, we do not create a property for the doc's embeddings
+            if column_name == self.embedding_column:
+                continue
+            prop = {
+                "name": column_name
+                if column_name != 'id'
+                else DOCUMENTID,  # in weaviate, id and _id is a reserved keyword
+                "dataType": column_info.config["dataType"],
+            }
+            properties.append(prop)
+
+        # TODO: What is the best way to specify other config that is part of schema?
+        # e.g. invertedIndexConfig, shardingConfig, moduleConfig, vectorIndexConfig
+        #       and configure replication
+        # we will update base on user feedback
+        schema["properties"] = properties
+        schema["class"] = self._db_config.index_name
+
+        # TODO: Use exists() instead of contains() when available
+        #       see https://github.com/weaviate/weaviate-python-client/issues/232
+        if self._client.schema.contains(schema):
+            logging.warning(
+                f"Found index {self._db_config.index_name} with schema {schema}. Will reuse existing schema."
+            )
+        else:
+            self._client.schema.create_class(schema)
+
+    @dataclass
+    class DBConfig(BaseDocumentIndex.DBConfig):
+        host: str = 'http://weaviate:8080'
+        index_name: str = 'Document'
+
+    @dataclass
+    class RuntimeConfig(BaseDocumentIndex.RuntimeConfig):
+        default_column_config: Dict[Type, Dict[str, Any]] = field(
+            default_factory=lambda: {
+                np.ndarray: {
+                    'dataType': ['number[]'],
+                },
+                docarray.typing.ID: {'dataType': ['string']},
+                bool: {'dataType': ['boolean']},
+                int: {'dataType': ['int']},
+                float: {'dataType': ['number']},
+                str: {'dataType': ['text']},
+                # `None` is not a Type, but we allow it here anyway
+                None: {},  # type: ignore
+            }
+        )
+
+        batch_config: Dict[str, Any] = field(
+            default_factory=lambda: DEFAULT_BATCH_CONFIG
+        )
+
+    def _del_items(self, doc_ids: Sequence[str]):
+        has_matches = True
+
+        operands = [
+            {"path": [DOCUMENTID], "operator": "Equal", "valueString": doc_id}
+            for doc_id in doc_ids
+        ]
+        where_filter = {
+            "operator": "Or",
+            "operands": operands,
+        }
+
+        # do a loop because there is a limit to how many objects can be deleted at
+        # in a single query
+        # see: https://weaviate.io/developers/weaviate/api/rest/batch#maximum-number-of-deletes-per-query
```

Comment:
Your choice of course, but imo it would also be ok to let the user run into that limit here